{
  "id": "EUVD-2020-0005",
  "enisaUuid": "37706cbc-2662-3293-a47d-0ac5a6afc771",
  "description": "A flaw was found in Ansible Engine when using Ansible Vault for editing encrypted files. When a user executes \\\"ansible-vault edit\\\", another user on the same computer can read the old and new secret, as it is created in a temporary file with mkstemp and the returned file descriptor is closed and the method write_data is called to write the existing secret in the file. This method will delete the file before recreating it insecurely. All versions in 2.7.x, 2.8.x and 2.9.x branches are believed to be vulnerable.",
  "datePublished": "Mar 16, 2020, 4:15:00 PM",
  "dateUpdated": "Jun 13, 2020, 4:15:00 AM",
  "baseScore": 3.9,
  "baseScoreVersion": "3.1",
  "baseScoreVector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:L/I:L/A:N",
  "references": [
    "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1740",
    "https://github.com/ansible/ansible/issues/67798",
    "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DKPA4KC3OJSUFASUYMG66HKJE7ADNGFW/",
    "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WQVOQD4VAIXXTVQAJKTN7NUGTJFE2PCB/",
    "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MRRYUU5ZBLPBXCYG6CFP35D64NP2UB2S/",
    "https://lists.debian.org/debian-lts-announce/2020/05/msg00005.html",
    "https://security.gentoo.org/glsa/202006-11",
    "https://github.com/advisories/GHSA-vcg8-98q8-g7mj",
    "https://nvd.nist.gov/vuln/detail/CVE-2020-1740",
    "https://github.com/ansible/ansible/commit/28f9fbdb5e281976e33f443193047068afb97a9b",
    "https://github.com/ansible/ansible/commit/2a563514f070a0a8ba64aebf6bce21194be96c73",
    "https://github.com/ansible/ansible/commit/685a4b6d3ff72186d2b4ffce73172a5446a71ccc",
    "https://github.com/ansible/ansible/commit/ef32a5bf96a89107986375516285253c1380d7ef",
    "https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-12.yaml",
    "https://www.suse.com/security/cve/CVE-2020-1740.html",
    "https://www.debian.org/security/2021/dsa-4950",
    "https://access.redhat.com/errata/RHSA-2020:1544",
    "https://access.redhat.com/errata/RHSA-2020:1543",
    "https://access.redhat.com/errata/RHSA-2020:1542",
    "https://access.redhat.com/errata/RHSA-2020:1541",
    "https://access.redhat.com/errata/RHBA-2020:1539",
    "https://access.redhat.com/errata/RHBA-2020:0547",
    "https://advisories.mageia.org/CVE-2020-1740.html",
    "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/WQVOQD4VAIXXTVQAJKTN7NUGTJFE2PCB/",
    "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/DKPA4KC3OJSUFASUYMG66HKJE7ADNGFW/",
    "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/MRRYUU5ZBLPBXCYG6CFP35D64NP2UB2S/"
  ],
  "aliases": [
    "CVE-2020-1740",
    "GHSA-vcg8-98q8-g7mj",
    "PYSEC-2020-12"
  ],
  "assigner": "redhat",
  "epss": 0.04,
  "enisaIdProduct": [
    {
      "id": "85ea62bf-a767-3c38-b03e-59d9f0cd2f3a",
      "product": {
        "name": "Ansible"
      },
      "product_version": "2.7.x, 2.8.x, 2.9.x"
    }
  ],
  "enisaIdVendor": [
    {
      "id": "aa775a90-3cdd-3e52-9539-6dcb8556ffe1",
      "vendor": {
        "name": "Red Hat"
      }
    }
  ],
  "cve": "CVE-2020-1740"
}